packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -2016,6 +2110,31 @@ export function pushStartInstance(
     }
   }
 
+  if (enableFloat) {
+    if (responseState.requiresEmbedding) {
+      responseState.requiresEmbedding = false;
+      if (__DEV__) {
+        // Dev only marker for later
+        (responseState: any).isDocumentEmbedded = true;
+      }
+      switch (type) {
+        case 'html': {
+          // noop
+          break;
+        }
+        case 'head':
+        case 'body': {
+          pushHtmlEmbedding(preamble, postamble, responseState);
+          break;
+        }
+        default: {
+          pushBodyEmbedding(target, postamble);
+          pushHtmlEmbedding(preamble, postamble, responseState);
+        }
+      }
+    }
```

Comment:
If would be nice if we didn't have to do this check on every dom element render. however it is currently modeled this way to keep the work in the render loop instead of the flush pass. I could move it to flush where we'd only need to special case completedRootSegment / partial-preamble branches which will get hit much less often.